test/services/common/translator/base_test_translator.py

```diff
@@ -47,37 +50,45 @@ def validate_service_translations(
     print()
     diffs = {}
     processing = ""
+
     try:
-        for desc, cim in kwargs.items():
-            processing = f"blank {desc}"
-            blank = type(cim)()
-
-            # Convert the blank object to protobuf and ensure it didn't get converted to an instance of PBIdentifiedObject,
-            # which indicates a missing `to_pb` implementation or import.
-            blank_as_pb = blank.to_pb()
-            assert type(blank_as_pb) is not PBIdentifiedObject, f"There is something wrong with {type(cim)}.to_pb. It is calling directly to the base class."
-
-            # noinspection PyUnresolvedReferences
-            translated_blank = service_type().add_from_pb(blank_as_pb)
-            assert translated_blank is not None, f"{blank_as_pb}: Failed to add the translated protobuf object to the service."
-            assert type(translated_blank) is type(cim), f"{translated_blank}: Converted object should be the same type as {cim}"
-
-            result = comparator.compare_objects(blank, translated_blank)
-            if result.differences:
-                diffs[f"blank {desc}"] = result
-
-            processing = f"populated {desc}"
-            _remove_unsent_references(cim)
-            # noinspection PyUnresolvedReferences
-            service = service_type()  # outside _add_with_unresolved_references so weak references on `cim` cant be garbage collected before being compared.
-            result = comparator.compare_objects(cim, _add_with_unresolved_references(service, cim))
-            if result.differences:
-                diffs[f"populated {desc}"] = result
+        for desc, _cim in types_to_test.items():
```

Comment:
suggestion: could rename `_cim` to `cim_builder` or similar to make it clearer that this isn't actually a cim class.